### ISSUE TITLE
False positives by bluetooth proximity

### ISSUE THREAD

**Issue by samiullah** [nic-delhi/AarogyaSetu_Android#366]
As an end user If my neighbour who lives around 50m away from tests positive, app shows that U r at risk. Since most of Smart phones are using Class B bluetooth modules, there must be a range class which filters proximity range to 6m only. So if any user comes in 6m range of patient he should be sent an alert of high risk.

**Comment by gulshan98125** [comment]
I can't seem to find the filter being applied to the bluetooth proximity detection in the source code too.

**Comment by samiullah** [comment]
They are using default range  of bluetooth of phone, that needs to be fixed

**Comment by gulshan98125** [comment]
Exactly! This has definitely created so many false positives. Threshold on RSSI could have reduced the false positives by a lot. I wonder how did they even released it without an RSSI cutoff!

**Comment by samiullah** [comment]
thanks @gulshan98125!  I have written 6m as there are some cases/research going on wherein they have found droplets can travel upto 20 feet.

**Comment by ritwikmishra** [comment]
@samiullah @gulshan98125 if the bluetooth scanning distance is 6m per device then bluetooth proximity per device becomes 12m (because both devices will be scanning in 6m area). Generally, one-floor height is ~3-4 meters. And since both the floors will have same Lat Long, will it give alert even if two people are separated by floors?

And I suppose if this indeed is the case
then it can start a chain a reaction where all the residents of a multi storey building will come in high risk, even if resident of one apartment gets positive (due to external factors).


> radio waves can still pass through objects, even concrete walls and floors, meaning that a Bluetooth signal is not limited to the room you’re in. - [source](https://www.bluetooth.com/blog/3-common-mythsabout-bluetooth/)

**Comment by gulshan98125** [comment]
@ritwikmishra Bluetooth proximity per device will still remain 6m and not 12m because a device will only detect other device placed at 6m. 
Also walls reduce signal by a lot so the signal received at 6m won't be the one equalivalent to in open space. Hence it will say that distance is >6m since distance is calculated on the basis of signal strength. 
About the chain reaction, it won't happen because Aarogya only informs another user when he comes in contact with covid+ and not with the one who was identified as risky by the app.

**Comment by ritwikmishra** [comment]
> Hence it will say that distance is >6m since distance is calculated on the basis of signal strength.

@gulshan98125 can you point me to the specific part in the source code where it calculates the distance on the basis of signal strength?
